vllm/model_executor/layers/quantization/compressed_tensors/scaled_mm_triton.py

```diff
@@ -0,0 +1,207 @@
+from typing import Optional, Type
+
+import torch
+import triton
+import triton.language as tl
+
+
+# This function handles some cases that can cause certain failure, e.g.
+# a tensor that has shape = (72, 48) but stride = (5120, 1).  It can happen,
+# for example by saving a tensor using torch.save() and then adjusting its
+# size afterwards and then trying to use it.  Unfortunately,
+# torch.is_contiguous() doesn't help since a transposed tensor doesn't return
+# True, even though it can be stored contiguously in memory.
```

Comment:
I thought it might have been occurring in vllm, but I haven't seen it happen.  I did it to myself when I was debugging though.  I did a torch.save() when I found a tensor that didn't have the correct output,  but the tensor was huge, so I changed the size of the tensor to (72, 48) while the old size was something like (131072, 5120).  However, the strides remained (5120, 1) and my kernel got the incorrect result, while torch.__int_mm() would get the correct result for torch._int_mm(a,b).  I was wondering why that was, so I looked at the pytorch code to figure it out and found out what they were doing.  I put it there, just in case, but I actually haven't seen this happen in vLLM though. I thought about removing the prepare_matrix_for_triton() function, but this scaled_mm_triton() function is general enough that it could be used elsewhere (it handles any input dtype and output dtype I've tried so far and can mix and match per-tensor, row-wise, and column-wise scaling), so I left it in.

Reply:
I will try taking this out and test, and put an assert where it was used instead, per suggestion on the other comment.

Reply:
Just using assert now.